components/src/AppModalBundleConfiguration/appmodalbundleconfiguration.main.tsx

```diff
@@ -94,21 +94,25 @@ class AppModalBundleConfigurationMain extends React.Component<AppModalBundleConf
     };
     this.handleErrorMessage = this.handleErrorMessage.bind(this);
     this.handleQuantityChange = this.handleQuantityChange.bind(this);
+    this.handleConfiguratorAddToCart = this.handleConfiguratorAddToCart.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
   }
 
   componentDidMount() {
-    this.fetchDependantItemData();
+    const { bundleConfigurationItems } = this.props;
+    this.fetchDependantItemData(bundleConfigurationItems);
   }
 
-  componentWillReceiveProps(nextProps) {
+
+  componentDidUpdate(prevProps) {
     const { bundleConfigurationItems } = this.props;
-    if (bundleConfigurationItems !== nextProps.bundleConfigurationItems) {
-      this.fetchDependantItemData();
-    }
+    if (JSON.stringify(bundleConfigurationItems) !== JSON.stringify(prevProps.bundleConfigurationItems)) {
```

Comment:
Would you happen to know why the object structure check is not valid here anymore? Just wondering why we didn't convert to strings to start with.

Reply:
It was never valid in the first place.  Javascript does not do a deep comparison of objects OOTB, instead it will compare the reference of objects.  In this case it always resulted in false and caused a re-render.  The condition was never doing anything.